wolfcrypt/src/cmac.c

```diff
@@ -326,14 +328,40 @@ int wc_AesCmacGenerate(byte* out, word32* outSz,
         return MEMORY_E;
     }
 #endif
+
 #ifdef WOLFSSL_CHECK_MEM_ZERO
     XMEMSET(((unsigned char *)cmac) + sizeof(Aes), 0xff,
         sizeof(Cmac) - sizeof(Aes));
     /* Aes part is checked by wc_AesFree. */
-    wc_MemZero_Add("wc_AesCmacGenerate cmac",
+    wc_MemZero_Add("wc_AesCmacGenerate_ex cmac",
         ((unsigned char *)cmac) + sizeof(Aes), sizeof(Cmac) - sizeof(Aes));
 #endif
 
+#ifdef WOLF_CRYPTO_CB
+    /* Set devId regardless of value (invalid or not) */
+    cmac->devId = devId;
+    #ifndef WOLF_CRYPTO_CB_FIND
+    if (devId != INVALID_DEVID)
+    #endif
+    {
+        cmac->devCtx = NULL;
+
+        ret = wc_CryptoCb_Cmac(cmac, key, keySz, in, inSz, out, outSz,
+                WC_CMAC_AES, NULL);
```

Comment:
Since [existing code in wc_InitCmac](https://github.com/wolfSSL/wolfssl/blob/b7b20ededda4cea208fb7745629904fda64c7524/wolfcrypt/src/cmac.c#L107) checks against WC_CMAC_AES for validity, I assumed it was supposed to hold a `CmacType` enum. And if we ever do CMAC with a different block cypher in the future, then you would multiplex between implementations based on that field. But our CMAC impl is hard-coded for AES at the moment so it is kind of pointless.

>Looking at the other callback invocations, it seems like it is set to 0 except during the init. 

I imagine this is just because only the init functions even take a type argument, so go unused during update/final

I can pass it 0 for this callback to maintain consistency